packages/core/src/StateMachine.ts

```diff
@@ -428,6 +448,74 @@ export class StateMachine<
     return resolvedState as State<TContext, TEvent, TResolvedTypesMeta>;
   }
 
+  /**
+   * Creates a new machine from this machine where the initial state is at the specified `state`.
+   *
+   * @param state The state to resolve the initial state from
+   */
+  public at(
+    state?:
+      | State<TContext, TEvent, TResolvedTypesMeta>
+      | PersistedMachineState<State<TContext, TEvent, TResolvedTypesMeta>>
+  ): this {
+    if (!state) {
+      return this;
+    }
+
+    const machine = new StateMachine(this.config) as typeof this;
+    let restoredState: State<TContext, TEvent, TResolvedTypesMeta>;
+
+    if ('persisted' in state) {
+      const restoredChildren: Record<string, AnyActorRef> = {};
+
+      Object.keys(state.children).forEach((key) => {
+        const persistedState = state.children[key];
+        const impl = this.options.actors[key];
+
+        if (!impl) return;
+
+        if (typeof impl === 'function') {
+          const behavior = impl(state.context, state.event, {
+            id: key,
+            src: {} as any,
+            _event: state._event,
+            meta: undefined
+          });
```

Comment:
This is a no-go - those arguments here are completely different from what usually can be used to create actors. This effectively makes typegen useless for actors because we'd have to annotate each actor factory with `TEvent` and not `TNarrowedEvent`. The context value is also way different here - it's used from a completely different moment in time. 

We try to re-instantiate "old" actors but we are using "new" values here. This is IMHO a conceptual clash

Reply:
What do you suggest we do instead? The actor implementations still expect those arguments, even if they're not used when restoring state.

Reply:
Ping @Andarist

Reply:
I'm still not sure how this can be fixed in this PR, since the solution depends on `input` which is in the linked PR. Open to any suggestions.

Also, I am open to disallowing `() => fromPromise()` completely 👍